Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -1235,20 +1271,26 @@ toggleDriverSubscriptionByService (driverId, mId, mOpCityId) serviceName mbPlanT
       _ <- DTPlan.planSubscribe serviceName planId (True, Just WHATSAPP) (cast driverId, mId, mOpCityId) driverInfo' serviceSpecificData
       pure ()
 
-runVerifyRCFlow :: Id DP.Person -> DM.Merchant -> Id DMOC.MerchantOperatingCity -> Common.AddVehicleReq -> Flow ()
-runVerifyRCFlow personId merchant merchantOpCityId req = do
+runVerifyRCFlow :: Id DP.Person -> DM.Merchant -> Id DMOC.MerchantOperatingCity -> Context.City -> Common.AddVehicleReq -> Bool -> Flow ()
+runVerifyRCFlow personId merchant merchantOpCityId operatingCity req isFleet = do
+  let vehicleCategory = case req.vehicleCategory of
+        Just category -> Just $ DashboardMerchant.castCategory category
+        Nothing -> Nothing
+  let imageId = maybe "" cast req.imageId
   let rcReq =
         DomainRC.DriverRCReq
           { vehicleRegistrationCertNumber = req.registrationNo,
-            imageId = "",
-            operatingCity = "Bangalore", -- TODO: this needs to be fixed properly
+            imageId = imageId,
+            operatingCity = show operatingCity, -- Fixed
             dateOfRegistration = Nothing,
-            vehicleCategory = Nothing,
             airConditioned = req.airConditioned,
             multipleRC = Nothing,
-            vehicleDetails = Nothing
+            vehicleDetails = Nothing,
+            vehicleCategory = vehicleCategory
           }
-  void $ DomainRC.verifyRC True (Just merchant) (personId, merchant.id, merchantOpCityId) rcReq
+  if isFleet
+    then void $ DomainRC.verifyRC False (Just merchant) (personId, merchant.id, merchantOpCityId) rcReq
+    else void $ DomainRC.verifyRC True (Just merchant) (personId, merchant.id, merchantOpCityId) rcReq
```

Comment:
`void $ DomainRC.verifyRC (not isFleet) (Just merchant) (personId, merchant.id, merchantOpCityId) rcReq`